tests/runtests.sh

```diff
@@ -75,18 +75,31 @@ ddimage ()
 	dd if=${imgsource} of=${imgname} bs=512 count=${blocks} > /dev/null 2>&1
 }
 
+createtap ()
+{
+	net=$1
+	ip link show ${net}
+	if [ $? -ne 0 ]; then
+		[ $(id -u) -ne 0 ] && die need root to create a tap
+		ip tuntap add ${net} mode tap
+		ip addr add 10.0.0.1/24 dev ${net}
+		ip link set dev ${net} up
+	fi
+}
+
 runguest ()
 {
 
 	testprog=$1
 	img1=$2
+	net1=$3
 	# notyet
 	# img2=$3
 
 	[ -n "${img1}" ] || die runtest without a disk image
 	case "${STACK}" in
 	spt)
-		cookie=$(${SOLO5_SPT} --disk=${img1} ${testprog} '{"cmdline":"testprog __test","blk":{"source":"etfs","path":"ld0d","fstype":"blk"}}')
+		cookie=$(${SOLO5_SPT} --block:rootfs=${img1} --net:tap=tap100 ${testprog} '{"cmdline":"testprog __test","blk":{"source":"etfs","path":"ld0d","fstype":"blk"}}')
```

Comment:
question. It won't be possible to run with a disk and not tap right?

Reply:
It is a current design issue. As I wrote in the comment, the rumprun system creates an object file from pre-defined `manifest.json` when running `build-rr.sh`. The gcc wrapper includes it when linking an application. The Solo5 tender expects all defined devices will be attached on run-time (Once a device is defined, it cannot be optional.) This is why the tests supply unnecessary devices.

Reply:
got it, thanks for the clarification.